tests/unit/test_reader_v1_arrow.py

```diff
@@ -0,0 +1,351 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import itertools
+from unittest import mock
+
+import pandas
+import pandas.testing
+import pytest
+
+import google.api_core.exceptions
+from google.cloud.bigquery_storage import types
+from .helpers import SCALAR_COLUMNS, SCALAR_COLUMN_NAMES, SCALAR_BLOCKS
+
+
+pyarrow = pytest.importorskip("pyarrow")
```

Comment:
This is the only new logic. Everything else is just moving from `test_reader_v1.py`